src/clj/coffi/mem.clj

```diff
@@ -1350,3 +1766,523 @@
+      (->> typelist
+           (map-indexed
+            (fn [index [offset [_ field-type]]]
+              (generate-serialize field-type (list (symbol (str "." (name (nth fieldnames index)))) 'source-obj) (if (number? global-offset) (+ global-offset offset) `(+ ~global-offset ~offset)) segment-source-form)))
+           (concat [`let ['source-obj source-form]])))))
+
+(gen-interface
+ :name coffi.mem.IStructImpl :methods
+ [[vec_length [] int]
+  [vec_assoc [Object Object] clojure.lang.Associative]
+  [vec_assocN [int Object] clojure.lang.IPersistentVector]
+  [vec_peek [] Object]
+  [vec_pop [] clojure.lang.IPersistentVector]
+  [vec_nth [int] Object]
+  [vec_nth [int Object] Object]
+  [vec_cons [Object] clojure.lang.IPersistentCollection]
+  [vec_equiv [Object] boolean]
+  [vec_empty [] clojure.lang.IPersistentVector]
+  [vec_iterator [] java.util.Iterator]
+  [vec_forEach [java.util.function.Consumer] void]
+  [vec_seq [] clojure.lang.ISeq]
+  [vec_rseq [] clojure.lang.ISeq]
+
+  [struct_count [] int]
+  [struct_containsKey [Object] boolean]
+  [struct_valAt [Object] Object]
+  [struct_valAt [Object Object] Object]
+  [struct_entryAt [Object] clojure.lang.IMapEntry]
+  [nthKey [int] clojure.lang.Keyword]
+
+  [map_assoc [Object Object] clojure.lang.Associative]
+  [map_assocEx [Object Object] clojure.lang.IPersistentMap]
+  [map_without [Object] clojure.lang.IPersistentMap]
+  [map_cons [Object] clojure.lang.IPersistentCollection]
+  [map_equiv [Object] boolean]
+  [map_empty [] clojure.lang.IPersistentMap]
+  [map_iterator [] java.util.Iterator]
+  [map_forEach [java.util.function.Consumer] void]
+  [map_seq [] clojure.lang.ISeq]
+  ;java.util.map fns
+  [map_containsValue [Object] boolean]
+  [map_entrySet [] java.util.Set]
+  [map_get [Object] Object]
+  [map_isEmpty [] boolean]
+  [map_keySet [] java.util.Set]
+  [map_size [] int]
+  [map_values [] java.util.Collection]
+  [map_forEach [java.util.function.BiConsumer] void]])
+
+
+(defmacro for-each-fixed-length [n]
+  `(defn ~(symbol (str "for-each-fixed-" n)) ~[(with-meta 'offset {:tag int}) (with-meta 'action {:tag 'java.util.function.Consumer}) (with-meta 's {:tag 'coffi.mem.IStructImpl})]
+     ~(cons `do (map (fn [i] (list '.accept (with-meta 'action {:tag 'java.util.function.Consumer}) (list '.vec_nth (with-meta 's {:tag 'coffi.mem.IStructImpl}) i))) (range n)))))
+
+(for-each-fixed-length 1)
+(for-each-fixed-length 2)
+(for-each-fixed-length 3)
+(for-each-fixed-length 4)
+(for-each-fixed-length 5)
+(for-each-fixed-length 6)
+(for-each-fixed-length 7)
+(for-each-fixed-length 8)
+(for-each-fixed-length 9)
+(for-each-fixed-length 10)
+(for-each-fixed-length 11)
+(for-each-fixed-length 12)
+(for-each-fixed-length 13)
+(for-each-fixed-length 14)
+(for-each-fixed-length 15)
+(for-each-fixed-length 16)
+
+(deftype struct-vec-iterator [^coffi.mem.IStructImpl struct-obj ^int size ^{:volatile-mutable true :tag int} i]
```

Comment:
Just for consistency this should probably be upper camelcase, like other types.
```suggestion
(deftype StructVecIterator [^coffi.mem.IStructImpl struct-obj ^int size ^{:volatile-mutable true :tag int} i]
```
This would also require another line in `generate-struct-type` to change the constructor call.

Reply:
I agree completely! fixed in [d24ebc7](https://github.com/IGJoshua/coffi/pull/14/commits/d24ebc7810a1c9b9b357173dee854384ac64657a)